lib/elixir/lib/task/supervised.ex

```diff
@@ -420,7 +420,18 @@ defmodule Task.Supervised do
         case running_tasks do
           %{^ref => {position, _type, pid, _timer_ref}} ->
             send(parent_pid, {:killed_for_timeout, {monitor_ref, position}})
+            caller = self()
+            ref = make_ref()
+            enforcer = spawn(fn ->
```

Comment:
If trapping exits we could skip spawning, unsure if worth complexity of that optimisation though.